ct-app/core/core.py

```diff
@@ -77,25 +86,46 @@ def network_nodes_addresses(self) -> list[Address]:
         return [node.address for node in self.network_nodes]
 
     @property
-    def safes_balance_subgraph_type(self) -> SubgraphType:
+    def subgraph_type(self) -> SubgraphType:
         return self._safes_balance_subgraph_type
 
-    @safes_balance_subgraph_type.setter
-    def safes_balance_subgraph_type(self, value: SubgraphType):
-        if value != self.safes_balance_subgraph_type:
+    @subgraph_type.setter
+    def subgraph_type(self, value: SubgraphType):
+        if value != self.subgraph_type:
             self.warning(f"Now using '{value.value}' subgraph.")
 
         SUBGRAPH_IN_USE.set(value.toInt())
         self._safes_balance_subgraph_type = value
 
-    def subgraph_safes_balance_url(self, subgraph: SubgraphType) -> str:
-        if subgraph == SubgraphType.DEFAULT:
+    @property
+    def subgraph_safes_balance_url(self) -> str:
+        if self.subgraph_type == SubgraphType.DEFAULT:
             return self.params.subgraph.safes_balance_url
 
-        if subgraph == SubgraphType.BACKUP:
+        if self.subgraph_type == SubgraphType.BACKUP:
             return self.params.subgraph.safes_balance_url_backup
 
-        return SubgraphType.NONE
+        return None
+
+    @property
+    def subgraph_staking_url(self) -> str:
+        if self.subgraph_type == SubgraphType.DEFAULT:
+            return self.params.subgraph.staking_url
+
+        if self.subgraph_type == SubgraphType.BACKUP:
+            return self.params.subgraph.staking_url_backup
+
+        return None
+
+    @property
+    def subgraph_wxhopr_txs_url(self) -> str:
+        if self.subgraph_type == SubgraphType.DEFAULT:
+            return self.params.subgraph.wxhopr_txs_url
+
+        if self.subgraph_type == SubgraphType.BACKUP:
+            return self.params.subgraph.wxhopr_txs_url_backup
+
+        return None
```

Comment:
As a general rule, this sort of mapping code is more practical as a const dict, e.g.:
```suggestion
        # the mapping can be a static const member of the class or module
        subgraph = {
            SubgraphType.DEFAULT: self.params.subgraph.wxhopr_txs_url,
            SubgraphType.BACKUP: self.params.subgraph.wxhopr_txs_url_backup
        }
        
        return subgraph.get(self.subgraph_type, None)
```